Application/EdFi.Ods.Common/Infrastructure/Interceptors/EdFiOdsInterceptor.cs

```diff
@@ -0,0 +1,69 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Text.RegularExpressions;
+using EdFi.Ods.Common.Conventions;
+using EdFi.Ods.Common.Security.CustomViewBased;
+using NHibernate;
+using NHibernate.SqlCommand;
+
+namespace EdFi.Ods.Common.Infrastructure.Interceptors
+{
+    public class EdFiOdsInterceptor : EmptyInterceptor
+    {
+        private const string SpaceLiteral = " ";
+
+        private static readonly Regex regex = new(
+            @$"(?<join>inner join\s+)\(?(?<basisEntity>\w+\.\w+) (?<alias>(?<aliasPrefix>{CustomViewHelpers.CustomViewAliasPrefixBase}[a-f\d]{{4}}_).*?\d_).*?(?<onClause>on\s+)",
+            RegexOptions.Compiled);
+
+        public override bool? IsTransient(object entity)
+        {
+            var property = entity.GetType()
+                                 .GetProperty("CreateDate");
+
+            if (property != null)
+            {
+                bool isTransient = Convert.ToDateTime(property.GetValue(entity)) == default(DateTime);
```

Comment:
I think the exception-handling infrastructure may need to be updated slightly to process an exception thrown from `property.GetValue()` (or possibly any exception thrown below the NHibernate interceptor invocation). From a quick check, it appears that any exception thrown in this context will be wrapped in a `GenericADOException` by NHibernate, which would prevent an Ed-Fi custom exception type from being correctly serialized into an API response. Also, having the outer exception type of `GenericADOException` showing first in logs/stack traces for cases where the exception is caused by a naming mismatch in the auth view configuration might make troubleshooting less straightforward.